woocommerce/woocommerce
# Add product deletion via datastore API

### All Submissions:






### Changes proposed in this Pull Request:



This PR adds product deletion via the datastore API.

Closes #33127.

### How to test the changes in this Pull Request:

1. Go to `wp-admin/edit.php?post_type=product`
2. Pick an ID of a recently published product
3. Open your browser’s console
4. Run the following.  You may need to run again after the fetch has been completed.
```
5. wp.data.select('wc/admin/products').getProducts({ status: 'publish'}, undefined);
```
6. Note the product you picked.
7. Enter the following, replacing the `my_product_id` ID with your product ID:
```
wp.data.dispatch('wc/admin/products').removeProduct( `my_product_id` );
```
8. Run the getProducts selector again:
```
wp.data.select('wc/admin/products').getProducts({ status: 'publish'}, undefined);
```
9. Note the product status now is `trash`.
10. Go to the `Trash` can (`/wp-admin/edit.php?post_status=trash&post_type=product`) and verify the product is there.
11. Now run the following line to delete the product definitively
```
wp.data.dispatch('wc/admin/products').removeProduct( `my_product_id`, { force: true } );
```
12. Run the getProducts selector again and note the product status now is `deleted`.
13. Go to the product list and the `Trash` can and verify the product is not present anymore.


### Other information:




### FOR PR REVIEWER ONLY:
